api/app/lib/statistics.rb

```diff
@@ -0,0 +1,104 @@
+require "google/apis/analytics_v3"
+require "google/api_client/auth/key_utils"
+
+class Statistics
+  include ActiveModel::Validations
+  include ActiveModel::Conversion
+  extend ActiveModel::Naming
+
+  def id
+    "statistics-set"
+  end
+
+  def initialize(_range = nil)
+    @client = Google::Apis::AnalyticsV3::AnalyticsService.new
+
+    @client.authorization = Signet::OAuth2::Client.new(
+      token_credential_uri:  "https://accounts.google.com/o/oauth2/token",
+      audience:              "https://accounts.google.com/o/oauth2/token",
+      scope:                 "https://www.googleapis.com/auth/analytics.readonly",
+      issuer:                 ENV["GOOGLE_SERVICE_CLIENT_EMAIL"],
+      signing_key:            OpenSSL::PKey::RSA.new(ENV["GOOGLE_SERVICE_PRIVATE_KEY"])
+    ).tap(&:fetch_access_token!)
+
```

Comment:
Move the creation of the authorization code to a factory model, similar to what I do for drive sessions. Setting up the session isn't the concern of the statistics model, per se, and we may have other cases outside of statistics where we need to get this token, so you may as well abstract it to its own class.